### DISCUSSION TITLE
Can't login on osu lazer

### DISCUSSION THREAD

**Discussion by Alexiiisv** [ppy/osu#15527] [Q&A]
When I try to login on osu lazer I receive this message :

`2021-11-08 17:40:50 [verbose]: Request to https://osu.ppy.sh/oauth/token failed with System.Net.WebException: BadRequest.
2021-11-08 17:40:50 [verbose]: Response was: {"error":"invalid_grant","error_description":"The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.","hint":"Incorrect sign in","message":"The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client."}
2021-11-08 17:40:50 [verbose]: Login failed!`

Tried on 2 different days with a username and email don't work. Even try to disconnect on the site and reconnect with the exact same username and password it work.

Running osu 2021.1108.0.0 on .NET 5.0.11

**Comment by bdach** [comment]
Are you sure you're logging in using your username and typing in the password correctly?

If yes, please check the `game.ini` file in your lazer install - does it have a line starting with `Token =`? If so, try removing everything in that one line after the `=` sign and saving it. (Probably wise to make a backup of the file before trying this.)

**Reply by Alexiiisv** [reply]
I use the same login and password for osu lazer and the website, only the website work.

About the `game.ini`, the line starting with `Token = ` after the `=` there is nothing written. Even when moving the line to the end of the file it doesn't work.

**Reply by peppy** [reply]
The error says that your username/password combination is not being accepted.

Can you try the following things in this order:

- Confirm whether your login works in stable (if you are able to run stable)
- Type your password in another window and paste into osu! to make sure it's not potentially being typed incorrectly where you can't see it.
-  Try changing your osu! password. Keep note of the previous one so we can potentially fix any issues specific to the password you were attempting.

Also please share your username and your osu! history (ie. have you been playing and using your account until now, or is this a new account?)

**Reply by Alexiiisv** [reply]
so I don't know exactly le problem but the only fix was to change my password.

I tried reconnecting in osu stable the connection failed. Even when typing my password in another window the error failed.

My username is alexisX2 I've been playing since september 2017.

**Reply by peppy** [reply]
If your previous password was unique to osu! and you're happy to share it, could you please do so? If you can't share the password, can you share the length and any weird characters or symbols it contained?

**Reply by Alexiiisv** [reply]
Is there any way I can share it in private ?

**Reply by peppy** [reply]
Yep, pe@ppy.sh works for that. Also, you could try changing your password back to the one you were using and see if the issue returns or not.

**Reply by Alexiiisv** [reply]
Mail send. I will not be able to try switching back to my old password rn trying when I'm back home.
